### PR TITLE
Ria 6605 ada suitability document api

### PR DESCRIPTION
**Pull request by hussain-faruk**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-6605

### Change description ###
Added ADASuitabilityReviewDecision entity. 
Added suitability review, reason and judge to AsylumCaseDefinition. 
Added AdaSuitability document tag. 
Added suitability review event. 
Created pre-submit handler to generate ada suitability document. 
Added AdaSuitability to bundleOrder.
Modified test for bundleOrder.
Created new AdaSuitabilityTemplate. 
Added Adasuitability bean to DocumentCreatorConfiguration. 
Added AdaSuitabilityDocument metadata to application.yml. Created AdaSuitabilityTemplateTest. 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Comment by hussain-faruk** [comment]
[DE 50002 2023-Dld-ADA-suitability-decision.PDF](https://github.com/hmcts/ia-case-documents-api/files/10511046/DE.50002.2023-Dld-ADA-suitability-decision.PDF)

Example ADA Suitability document attached

File: src/main/resources/application.yaml [hunk 1]
```diff
@@ -99,6 +99,11 @@ adaHearingNoticeUpdatedDetailsDocument.fileExtension: PDF
 adaHearingNoticeUpdatedDetailsDocument.fileName: "hearing-notice"
 adaHearingNoticeUpdatedDetailsDocument.templateName: ${IA_ADA_HEARING_NOTICE_UPDATED_DETAILS_TEMPLATE:TB-IAC-HNO-ENG-ADA-0002.docx}
 
+AdaSuitabilityDocument.contentType: application/pdf
```
Comment (ThomasKKC) [inline src/main/resources/application.yaml:102]: Can you make the first char lower case like the rest?

File: src/test/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/templates/AdaSuitabilityTemplateTest.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    private AdaSuitabilityReviewDecision adaSuitability = AdaSuitabilityReviewDecision.UNSUITABLE;
+    private String adaSuitabilityReason = "This is the reason";
+    private String adaSuitabilityjudge = "Judgy Judgerson";
+
+
+    @BeforeEach
+    void setUp() {
+
+
+
```
Comment (ThomasKKC) [inline src/test/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/templates/AdaSuitabilityTemplateTest.java:51]: Can you fix the formatting here?

**Review by ThomasKKC — CHANGES_REQUESTED** [review]

**Review by ThomasKKC — CHANGES_REQUESTED** [review]
We need to add functional tests for new template/scenario

**Review by ThomasKKC — APPROVED** [review]

**Review by colmlynch60 — APPROVED** [review]
LGTM - Thomas and Hussain has brought me through the required changes here

**Pull request merged**